pkg/products/rhssouser/reconciler.go

```diff
@@ -1016,3 +1032,188 @@ func (r *Reconciler) reconcileBrowserAuthFlow(ctx context.Context, kc *keycloak.
+			return kcClient.CreateGroupClientRole(role, masterRealmName, clientID, groupID)
+		},
+	)
+}
+
+func mapRealmRoleToGroup(kcClient keycloakCommon.KeycloakInterface, groupID, roleName string) error {
+	return mapRoleToGroupByName(roleName,
+		func() ([]*keycloak.KeycloakUserRole, error) {
+			return kcClient.ListGroupRealmRoles(masterRealmName, groupID)
+		},
+		func() ([]*keycloak.KeycloakUserRole, error) {
+			return kcClient.ListAvailableGroupRealmRoles(masterRealmName, groupID)
+		},
+		func(role *keycloak.KeycloakUserRole) error {
+			return kcClient.CreateGroupRealmRole(role, masterRealmName, groupID)
+		},
+	)
+}
+
+// Map a role to a group given the name of the role to map and the logic to:
+// list the roles already mapped, list the roles available to the group, and
+// to map a role to the group
+func mapRoleToGroupByName(
+	roleName string,
+	listMappedRoles func() ([]*keycloak.KeycloakUserRole, error),
+	listAvailableRoles func() ([]*keycloak.KeycloakUserRole, error),
+	mapRoleToGroup func(*keycloak.KeycloakUserRole) error,
+) error {
+	// Utility local function to look for the role in a list
+	findRole := func(roles []*keycloak.KeycloakUserRole) *keycloak.KeycloakUserRole {
+		for _, role := range roles {
+			if role.Name == roleName {
+				return role
+			}
+		}
+
+		return nil
+	}
+
+	// Get the existing roles mapped to the group
+	existingRoles, err := listMappedRoles()
+	if err != nil {
+		return err
+	}
+
+	// Look for the role among them, if it's already there, return
+	existingRole := findRole(existingRoles)
+	if existingRole != nil {
+		return nil
+	}
+
+	// Get the available roles for the group
+	availableRoles, err := listAvailableRoles()
+	if err != nil {
+		return err
+	}
+
+	// Look for the role among them. If it's not found, return an error
+	role := findRole(availableRoles)
+	if role == nil {
+		return fmt.Errorf("%s role not found as available role for group", roleName)
+	}
+
+	// Map the role to the group
+	return mapRoleToGroup(role)
+}
+
+func (r *Reconciler) reconcileFirstLoginAuthFlow(kc *keycloak.Keycloak) (integreatlyv1alpha1.StatusPhase, error) {
+	// Get Keycloak client
+	kcClient, err := r.keycloakClientFactory.AuthenticatedClient(*kc)
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	// Find the "review profile" execution for the first broker login
+	// authentication flow
+	authenticationExecution, err := kcClient.FindAuthenticationExecutionForFlow(firstBrokerLoginFlowAlias, masterRealmName, func(execution *keycloak.AuthenticationExecutionInfo) bool {
+		return execution.Alias == reviewProfileExecutionAlias
+	})
+	if err != nil {
+		return integreatlyv1alpha1.PhaseFailed, err
+	}
+
+	// If the execution is not found, nothing needs to be done
+	if authenticationExecution == nil {
+		return integreatlyv1alpha1.PhaseCompleted, nil
+	}
+	// If the execution is already disabled, nothing needs to be done
+	if strings.ToLower(authenticationExecution.Requirement) == "DISABLED" {
```

Comment:
@sergioifg94 we are changing string to lower case and comparing with upper case one, that will never match.

Reply:
Meant to call `ToUpper` there, I'll fix it now